src/core/evaluator.js

```diff
@@ -1676,13 +1689,19 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
               }
               break;
           } // switch
+          if (textContent.items.length >= sink.desiredSize) {
+            // wait for ready, if we reach highWaterMark.
+            stop = true;
+            break;
+          }
         } // while
         if (stop) {
           next(deferred);
           return;
         }
         flushTextContentItem();
-        resolve(textContent);
+        enqueueChunk();
+        resolve();
```

Comment:
Similar changes needs to be made in the `if (this.options.ignoreErrors) {` block of the `catch` below, since otherwise you're definitely going to break text-extraction in some PDF files.

*Edit:* The `return textContent;` line probably needs to replaced with
```javascript
enqueueChunk();
return;
```